crates/viewer/re_data_ui/src/image.rs

```diff
@@ -66,7 +66,11 @@ pub fn texture_preview_ui(
     texture: ColormappedTexture,
 ) -> egui::Response {
     if ui_layout.is_single_line() {
-        let preview_size = Vec2::splat(ui.available_height());
+        let height = ui.available_height();
+        let [texture_width, texture_height] = texture.width_height();
+        let width =
+            (height * texture_width as f32 / texture_height as f32).at_most(ui.available_width());
+        let preview_size = Vec2::new(width, height);
```

Comment:
This will break the aspect ratio if the `available_width()` is less than the required width. 
I think we should aim to maintain aspect ratio by computing a scale factor for both width and height and picking the minimum value so we don't overflow either dimension, e.g.
```suggestion
        let [texture_width, texture_height] = texture.width_height();

        let available_height = ui.available_height();
        let available_width = ui.available_width();

        let scale_height = available_height / texture_height as f32;
        let scale_width = available_width / texture_width as f32;

        let scale = scale_height.min(scale_width);

        let preview_size = Vec2::new(texture_width as f32 * scale, texture_height as f32 * scale);
```

Reply:
egui will handle the aspect ratio, so this should be fine